### DISCUSSION TITLE
788

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#960] [Comments]
# 788

PostgreSQL 14 에서 진행되었다.간혹 쿼리들을 보면 단일 테이블 (from 테이블)의 컬럼으로 모든 조회 조건이 완성 가능한데, join 테이블 의 조건을 함께 사용하여 성능 저하를 일으키는 경우가 종종 있다.데이터가 몇개 없을때는 큰 차이를 못 느끼지만, 수십만건 이상이 적재된 여러 테이블을 중첩 Join 할 경우 큰 차이가 느껴지게 된다.이를 비교해보자.문제아래의 쿼리는 review 테이블과 연관된 여러 테이블의 정보를 모으고, 이를 페이징 처리하여 제공해야하는 기능이다.select *from "review" as reviewleft join "users" as "user" on review."user_id" = "user"."id"left join "courses" as course on ..

https://jojoldu.tistory.com/788

**Comment by mystyle2006** [comment]
저에게는... 실행 계획을 해보지 않았다면 배울 수 없었던 성능 개선이었습니다..

**Reply by mystyle2006** [reply]
리뷰에 대한 스캔 방식부터 달라지네..
